File: paths.go
package main

import (
	"errors"
	"os"
	"path/filepath"
)

func ResolveAppDir() (string, error) {
	exePath, err := os.Executable()
	if err != nil {
		return "", err
	}
	if exePath == "" {
		return "", errors.New("executable path is empty")
	}
	return filepath.Dir(exePath), nil
}

File: src/backend/paths.rs
use std::io;
use std::path::PathBuf;

pub fn resolve_app_dir() -> io::Result<PathBuf> {
    let exe = std::env::current_exe()?;
    exe.parent()
        .map(PathBuf::from)
        .ok_or_else(|| io::Error::other("executable has no parent directory"))
}
